components/amqp/sources/receive-message/receive-message.mjs

```diff
@@ -0,0 +1,83 @@
+import amqp from "../../amqp.app.mjs";
+
+export default {
+  key: "amqp-receive-message",
+  name: "New Message",
+  description: "Emit new event for each new message in an AMQP 1.0 queue",
+  type: "source",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    amqp,
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    receiverName: {
+      type: "string",
+      label: "Receiver Name",
+      description: "The name of the receiver. e.g. (`my-receiver`)",
+    },
+    queueName: {
+      propDefinition: [
+        amqp,
+        "queueName",
+      ],
+    },
+  },
+  async run(event) {
+    const { timestamp } = event;
+    const {
+      host,
+      port,
+      username,
+      password,
+    } = this.amqp.$auth;
+
+    const {
+      receiverName,
+      queueName,
+    } = this;
+
+    console.log("Open connection");
+    const connection = await this.amqp.openConnection({
+      host,
+      port,
+      username,
+      password,
+    });
+
+    console.log("Create receiver");
+    const receiver = await this.amqp.createReceiver({
+      connection,
+      name: receiverName,
+      source: {
+        address: queueName,
+      },
+      onSessionError: (context) => {
+        throw context.session.error ?? "Unknown Session Error";
+      },
+    });
+
+    try {
+      const message = await this.amqp.onMessageReceiver(receiver);
+
+      const id = Date.parse(timestamp);
+
+      this.$emit(message, {
+        id,
+        ts: id,
+        summary: `New Message with ID ${message.message_id}`,
+      });
+
+    } catch (error) {
+      console.log("Error receiving message", JSON.stringify(error));
```

Comment:
We don't wanna throw the error? Looks like if some error occurs we will just console. Maybe we could console and `throw error`?